### PR TITLE
Fix #3866 - Added IPv6 support to the URL bar

### LINKED ISSUE #3866: IPv6 addresses not identified as valid URLs
See upstream ticket

https://github.com/mozilla-mobile/firefox-ios/issues/6015

### PR DESCRIPTION
**Pull request by Brandon-T**
## Summary of Changes
- Security Review: https://github.com/brave/security/issues/516
- Added IPAddress validation and allow typing in VALID IPv6 addresses into the URL bar.

This pull request fixes #3866
Addresses the same FF issue: mozilla-mobile/firefox-ios#6015

## Submitter Checklist:

- [X] *Unit Tests* are updated to cover new or changed functionality
- [X] User-facing strings use `NSLocalizableString()`

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

**Review by kylehickinson — APPROVED** [review]

File: Client/Frontend/Browser/URIFixup.swift [hunk 1]
```diff
@@ -18,6 +38,12 @@ class URIFixup {
             if decodedASCIIURL.rangeOfCharacter(from: CharacterSet.URLAllowed.inverted) != nil {
                 return nil
             }
+            
+            // The host is a valid IPv4 or IPv6 address
+            var buffer = [UInt8](repeating: 0, count: Int(INET6_ADDRSTRLEN))
+            if inet_pton(AF_INET, host, &buffer) != 0 || inet_pton(AF_INET6, host, &buffer) != 0 {
```
Comment (jumde) [inline Client/Frontend/Browser/URIFixup.swift]: nit: can we convert this duplicated code into a utility function?

File: Client/Frontend/Browser/URIFixup.swift [hunk 2]
```diff
@@ -6,6 +6,26 @@ import Foundation
...
+            }
+            return false
+        }
+        
+        if let url = URL(string: "https://\(string)"),
+           let host = url.host, !host.isEmpty,
+           url.port != nil {
+            return isValidIPAddress(host)
+        } else if URL(string: "https://\(string)") != nil || URL(string: string) != nil {
+            return isValidIPAddress(string)
```
Comment (jumde) [inline Client/Frontend/Browser/URIFixup.swift]: `[::192.9.5.5]/ipng` fails the `isValidIPAddress` URL check - URL from the RFC: https://www.ietf.org/rfc/rfc2732.txt

Comment (Brandon-T) [reply Client/Frontend/Browser/URIFixup.swift]: Fixed.

**Review by soner-yuksel — APPROVED** [review]
LGTM nice clean tests

**Review by jumde — APPROVED** [review]
++

**Review by stonigrl — COMMENTED** [review]
👌

**Pull request merged**